packages/page-bounties/src/ExtendBountyExpiryAction.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017-2021 @polkadot/app-treasury authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { AccountId, BountyIndex } from '@polkadot/types/interfaces';
+
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Button, Input, InputAddress, Modal, TxButton } from '@polkadot/react-components';
+import { useAccounts, useToggle } from '@polkadot/react-hooks';
+
+import { truncateTitle } from './helpers';
+import { useBounties } from './hooks';
+import { useTranslation } from './translate';
+
+interface Props {
+  description: string
+  index: BountyIndex;
+  curatorId: AccountId;
+}
+
+function ExtendBountyExpiryAction ({ curatorId, description, index }: Props): React.ReactElement<Props> | null {
+  const { t } = useTranslation();
+  const { extendBountyExpiry } = useBounties();
+  const [isOpen, toggleOpen] = useToggle();
+  const { allAccounts } = useAccounts();
+  const [remark, setRemark] = useState('');
+  const isCurator = useMemo(() => allAccounts.includes(curatorId.toString()), [allAccounts, curatorId]);
+
+  const onRemarkChange = useCallback((value: string) => {
+    setRemark(value);
+  }, []);
+
+  return isCurator
+    ? (
+      <>
+        <Button
+          icon='calendar-plus'
+          isDisabled={false}
+          label={t<string>('Extend Bounty Expiry')}
```

Comment:
```suggestion
          label={t<string>('Extend Expiry')}
```

We know we are working with a bounty here.